src/Symfony/Component/Messenger/Transport/Semaphore/Connection.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Semaphore;
+
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Component\Messenger\Transport\Semaphore\Exception\SemaphoreException;
+use Symfony\Component\Messenger\Transport\Semaphore\Util\PlatformUtil;
+
+/**
+ * A Semaphore connection.
+ *
+ * @author Cedrick Oka Baidai <okacedrick@gmail.com>
+ */
+class Connection
+{
+    private const DEFAULT_OPTIONS = [
+        'path' => __FILE__,
+        'project' => 'M',
+        'message_max_size' => 131072,
+        'auto_setup' => true,
+    ];
+
+    /**
+     * @var array
+     */
+    private $configuration;
+
+    /**
+     * @var resource
+     */
+    private $queue;
+
+    /**
+     * @var bool
+     */
+    private $setup;
+
+    public function __construct(array $configuration)
+    {
+        if (true === PlatformUtil::isWindows()) {
```

Comment:
Shouldn't we rather check `\extension_loaded('sysvsem')`, as for the Lock component `SemaphoreStore`?

https://github.com/symfony/symfony/blob/d2649f2f7845cf40dec7c0f44704382b99d30c56/src/Symfony/Component/Lock/Store/SemaphoreStore.php#L36-L41